src/field.cpp

```diff
@@ -2071,6 +2096,44 @@ void map::player_in_field( player &u )
             }
             break;
 
+        case fd_fungicidal_gas:
+            // Fungicidal gas is unhealthyand becomes deadly if you cross a related threshold.
+            {
+                if (inside) break;
+                // The gas won't harm you inside a vehicle.
+                bool inhaled = false;
+                // Full body suits protect you from the effects of the gas.
+                if( !u.worn_with_flag("GAS_PROOF") ) {
+                    if( cur->getFieldDensity() == 3 ) {
```

Comment:
Invert those `if`s and make them more like a formula. Field code is currently quite bad with the whole `if( density == 3) { something; } else if( density == 2 )` type coding, no need to make it worse.

It could work like this: first handle the effects that happen to everyone (inhaled poison), like this:
```
const int density = cur->getFieldDensity();
inhaled = u.add_env_effect( "poison", bp_mouth, 5, density * 10 );
```
Then just the fungus stuff:
```
if( u. fungus check here ) {
    inhaled |= u.add_env_effect( "badpoison", bp_mouth, 5, density * 10 );
    u.hurtall( rng( density, density * 2 ), nullptr );
    u.add_msg_if_player( m_bad, _("hurts")...
}
```
Note that `|=`. If you use just the `=`, partially resisting the gas would give no message. Regular poison inhalation could set it to true, then bad poison could set it back to false if resisted.

No need to have separate messages for burns. If you can get them "for free" it's fine, but here it would look cleaner without a big if/else block.